lib/api/core/models/repositories/repository.js

```diff
@@ -59,19 +60,28 @@ class Repository {
    * @param {object} options
    */
   init (options) {
-    // if set to null, it means that we explicitely dont want a databaseEngine for this repository
-    // as null is falsy, without this check, the databaseEngine is always set...
-    if (options.databaseEngine === null) {
-      this.databaseEngine = options.databaseEngine;
+    // if set to null, it means that we explicitely dont want a indexEngine for this repository
+    // as null is falsy, without this check, the indexEngine is always set...
+    if (options.indexEngine === null) {
+      this.indexEngine = options.indexEngine;
     }
     else {
-      this.databaseEngine = options.databaseEngine || this.kuzzle.internalEngine;
+      // @todo refactor storageEngine set (it's dirty to pass it with options.indexEngine)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)